shared/middleware/net_sim_api.rb

```diff
@@ -328,14 +344,24 @@ def validate_wire(shard_id, wire)
     dont_cache
     unsupported_media_type unless has_json_utf8_headers?(request)
 
+    body_string = request.body.read
+
+    # Block request if payload is unusually large.
+    if body_string.length > MAX_REQUEST_SIZE
+      record_metric("UpdateTooLarge_#{table_name}", body_string.length)
+      too_large
+    end
```

Comment:
This approach may turn out to be hard to monitor. The easiest two ways I know of to monitor new relic metrics are via custom dashboards:
* old way: https://rpm.newrelic.com/accounts/501463/custom_dashboards/pages/7067
* new way (Insights): https://insights.newrelic.com/accounts/501463/dashboards/407434

The problem is that both prompt you to specify the custom metric by name, so if your metric name varies, then I'm not sure how you'll find it. Some possible solutions:
1. make this a custom event instead of a custom metric, which lets you include as many fields as you want, and can be searched using the [events explorer](https://insights.newrelic.com/accounts/501463/explorer/events) (though may not keep data as long)
2. use the [metrics explorer](https://insights.newrelic.com/accounts/501463/explorer/metrics) and rely on the autocomplete there to show which metrics have values (e.g. [metrics explorer for Custom/FilesApi](https://insights.newrelic.com/accounts/501463/explorer/metrics#entityId=3787364&entityName=Dashboard&searchTerms=%5ECustom%2FFilesApi%2F&entityType=application&entityTypeName=Applications&view=METRICS_OF_ENTITY&timeWindow=30))
3. also increment a counter that shows you the total number of `InsertTooLarge` and `UpdateTooLarge` events, add that a dashboard, and only drill down when that number goes above zero
4. if you know all the `table_name`s and there aren't too many, you could add each `#{metric_name}_#{table_name}` combination to a single dashboard

Reply:
So in total I'm introducing 21 new metrics:

- (1) `Custom/NetSimApi/ResetShard`
- (4) `Custom/NetSimApi/InsertBytes_(n|w|m|l)`
- (4) `Custom/NetSimApi/InsertRows_(n|w|m|l)`
- (4) `Custom/NetSimApi/UpdateBytes_(n|w|m|l)`
- (4) `Custom/NetSimApi/InsertTooLarge_(n|w|m|l)`
- (4) `Custom/NetSimApi/UpdateTooLarge_(n|w|m|l)`

Reply:
Ahh ok, the parameterized custom metric names make great sense then. Those should all fit nicely in a dashboard or two.